pkg/name/registry_test.go

```diff
@@ -108,6 +108,24 @@ func TestDefaultRegistryNames(t *testing.T) {
 	}
 }
 
+func TestOverrideDefaultRegistryNames(t *testing.T) {
+	testRegistries := []string{"docker.io", ""}
+	expectedRegistries := []string{"index.docker.io", "gcr.io"}
+	overrideDefault := "gcr.io"
```

Comment:
What happens if this is `gcr.io/myproject`?

Reply:
It would use `gcr.io/myproject`. We do not check the default, so a user has the ability to provide something nonsensical there. We were not previously validating the default either, but there was also only one option that could ever be the default. I could go either way on this, but I lean towards letting the user do what they want if they are opting to override. Typically I imagine the override will be hardcoded, rather than changing on accepted input, so the package consumer is responsible for making sure they use a reasonable value. What do you think?

Reply:
> I could go either way on this, but I lean towards letting the user do what they want if they are opting to override.

Yeah there is precedent for this (repo.Tag(), repo.Digest()), so I think it's fine.

I'd like to see tests for this that actually calls `ParseReference` instead of `NewRegistry` (basically nobody does that). A list of triples with (input, defaultRegistry, output) that includes this case would be great.

Reply:
@jonjohnsonjr SGTM. Would you be open to these tests being refactored to table-driven in a follow-up PR?

Reply:
@jonjohnsonjr I have updated with the `ParsseReference` tests here :+1:

Reply:
> Would you be open to these tests being refactored to table-driven in a follow-up PR?

Yeah that's fine, honestly this whole package needs that treatment and some docs love.

Reply:
@jonjohnsonjr sounds good! I'll get to work on that next 👍 Any more updates needed here?